### PR TITLE
add initial migration guide

### PR DESCRIPTION
**Pull request by ISO50**
close #158

### CODE REVIEW

File: MIGRATION_GUIDE.md [hunk 1]
```diff
@@ -2,7 +2,42 @@
 
 ## Migrate from `authmosphere@1.0.x` to `authmosphere@2.0.y`
 
-**TODO**
+### Upgrade
+
+* call `npm install --save authmosphere@~2.0.0`
```
Comment (bzums) [inline MIGRATION_GUIDE.md]: rather run? ;)

File: MIGRATION_GUIDE.md [hunk 2]
```diff
@@ -2,7 +2,42 @@
...
+
+### General changes
+
+* All exported functions got support for a custom Logger. Providing a logger is optional.
```
Comment (bzums) [inline MIGRATION_GUIDE.md]: "custom _l_ogger"

File: MIGRATION_GUIDE.md [hunk 3]
```diff
@@ -2,7 +2,42 @@
...
+  * an optional logger can be provided ([`Logger`](./src/types/Logger.ts))
+  * an optional `onNotAuthenticatedHandler` can be provided, it helps to customize handling the case authentication fails
+
+* `requireScopesMiddleware`
+  * added optional `options` object of type [`ScopeMiddlewareOptions`](./src/types/ScopeMiddlewareOptions.ts)
+    * an optional logger can be provided ([`Logger`](./src/types/Logger.ts))
+    * an optional `onAuthorizationFailedHandler` can be provided, it helps to customize handling the case authentication fails
+  * moved `precedenceOptions` parameter into `options` parameter
+    * `precedenceErrorHandler` got removed from [`PrecedenceOptions`](./src/types/Precedence.ts).
+      `onAuthorizationFailedHandler` should be used instead.
```
Comment (bzums) [inline MIGRATION_GUIDE.md]: Is "instead" really correct here, isnt it a different use case?

File: MIGRATION_GUIDE.md (hunk 1 above)
Comment (ISO50) [reply MIGRATION_GUIDE.md]: much better

**Comment by bzums** [comment]
👍

File: MIGRATION_GUIDE.md (hunk 3 above)
Comment (ISO50) [reply MIGRATION_GUIDE.md]: Differences will be explained in the documentation

**Comment by ISO50** [comment]
👍

**Pull request merged**